src/erlcloud_sm.erl

```diff
@@ -0,0 +1,136 @@
+-module(erlcloud_sm).
+-author("joshua@halloapp.com").
+
+-include("erlcloud.hrl").
+-include("erlcloud_aws.hrl").
+
+%%% Library initialization.
+-export([new/2, new/3, new/4]).
+
+%%% API
+-export([
+    get_secret_value/2, get_secret_value/3
+]).
+
+%%%------------------------------------------------------------------------------
+%%% Shared types
+%%%------------------------------------------------------------------------------
+
+-type return_type() :: {ok, proplists:proplist()} | {error, term()}.
+
+-type get_secret_value_option() :: {version_id | version_stage, binary()}.
+-type get_secret_value_options() :: [get_secret_value_option()].
+
+%%%------------------------------------------------------------------------------
+%%% Library initialization.
+%%%------------------------------------------------------------------------------
+
+-spec new(string(), string()) -> aws_config().
+new(AccessKeyID, SecretAccessKey) ->
+    #aws_config{
+        access_key_id = AccessKeyID,
+        secret_access_key = SecretAccessKey
+    }.
+
+
+-spec new(string(), string(), string()) -> aws_config().
+new(AccessKeyID, SecretAccessKey, Host) ->
+    #aws_config{
+        access_key_id = AccessKeyID,
+        secret_access_key = SecretAccessKey,
+        sm_host = Host
+    }.
+
+
+-spec new(string(), string(), string(), non_neg_integer()) -> aws_config().
+new(AccessKeyID, SecretAccessKey, Host, Port) ->
+    #aws_config{
+        access_key_id = AccessKeyID,
+        secret_access_key = SecretAccessKey,
+        sm_host = Host,
+        sm_port = Port
+    }.
+
+%%------------------------------------------------------------------------------
+%% GetSecretValue
+%%------------------------------------------------------------------------------
+%% @doc
+%% SM API:
+%% [https://docs.aws.amazon.com/secretsmanager/latest/apireference/API_GetSecretValue.html]
+%% @end
+%%------------------------------------------------------------------------------
+
+-spec get_secret_value(SecretId :: binary(), Opts :: get_secret_value_options()) -> return_type().
+get_secret_value(SecretId, Opts) ->
+    get_secret_value(SecretId, Opts, erlcloud_aws:default_config()).
+
+
+-spec get_secret_value(SecretId :: binary(), Opts :: get_secret_value_options(),
+        Config :: aws_config()) -> return_type().
+get_secret_value(SecretId, Opts, Config) ->
+    Json = lists:map(
+        fun({Version, Val}) ->
+            case Version of
```

Comment:
It would be more compact/readable to just use function heads here, I think:

```erlang
fun({version_id, Val}) -> {<<"VersionId">>, Val};
   ({version_stage, Val} -> {<<"VersionStage">>, Val});
   (Other) -> error(invalid_option, [Other])
end
```

What you do in the `Other` case is up to you. Supporting passthroughs could be useful, or you could be pedantic and explode on options you don't have mappings for.

Reply:
I agree, this is definitely more compact and readable. I also chose to allow any Other value to pass through without an error, as I pass the `SecretId` option through this function as well